examples/cifar10_cnn.py

```diff
@@ -99,3 +109,37 @@
                         steps_per_epoch=x_train.shape[0] // batch_size,
                         epochs=epochs,
                         validation_data=(x_test, y_test))
+
+# Save model and weights
+ensure_dir_exists(save_dir)
+model.save(save_dir+model_name)
+model.save_weights(save_dir+weights_name)
+print('Saved trained model: ' +save_dir+model_name)
+print('Saved weights: ' +save_dir+weights_name)
+
+# Load label names to use in prediction results
+label_list = os.path.expanduser("~") + '/.keras/datasets/cifar-10-batches-py/batches.meta'
+with open(label_list, mode='rb') as f:
+    labels = pickle.load(f)
+
+# Evaluate model with test data set and share sample prediction results
+evaluation = model.evaluate_generator(datagen.flow(x_test, y_test,
+                                      batch_size=batch_size),
+                                      steps=x_test.shape[0] // batch_size)
+
+eval_results = map(lambda metric, result: metric + ' = ' + str(result), \
+                    model.metrics_names, evaluation)
+print("Evaluation results: " + '  '.join(eval_results))
+
+
+predict_y = model.predict_generator(datagen.flow(x_test, y_test,
+                                    batch_size=batch_size),
+                                    steps=x_test.shape[0] // batch_size)
+
+for idx, y in enumerate(predict_y):
+  print("Actual | Predicted = %s | %s"% (labels['label_names'][np.argmax(y_test[idx])], \
+                                          labels['label_names'][np.argmax(y)]))
+  if idx == num_pred_print: break
```

Comment:
Don't put logic on an `if` line, introduce a line break. In general, I suggest rewriting the part of the code that prints evaluation results to make it simpler, easier to read, and more user friendly. A big goal of these example scripts is to be as user friendly as possible while introducing best practices.